src/GogeDao.sol

```diff
@@ -370,718 +476,955 @@ contract GogeDAO is Ownable {
+    /// @param  _pollNum unique poll identifier.
+    /// @dev    poll must be an active poll
+    function passPoll(uint256 _pollNum) external onlyOwner() {
+        require(isActivePoll(_pollNum), "Poll is not active");
+        _executeProposal(_pollNum);
+    }
+
+    /// @notice An owner method for updating status of createPoll.
+    function toggleCreatePollEnabled() external onlyOwner() {
+        createPollEnabled = !createPollEnabled;
+    }
+
+    /// @notice An owner method for manually ending a poll.
+    /// @param  _pollNum unique poll identifier.
+    /// @dev    Poll must be an active poll.
+    ///         This function is also callable by the author of _pollNum.
+    function endPoll(uint256 _pollNum) external onlyOwnerOrAuthor(_pollNum) {
+        require(isActivePoll(_pollNum), "Poll is not active");
+        _updateEndTime(_pollNum);
+        _removePollFromActivePolls(_pollNum);
+        _refundVotersPostChange(_pollNum);
+    }
+
+    /// @notice An owner method for updating minPollPeriod.
+    /// @param  _amount new minPollPeriod.
+    function updateMinPollPeriod(uint256 _amount) external onlyOwner() {
+        _updateMinPollPeriod(_amount);
+    }
+
+    /// @notice An owner method for adding new team member.
+    /// @param  _account new team member.
+    /// @param  _isMember is a team member.
+    function setTeamMember(address _account, bool _isMember) external onlyOwner() {
+        _setTeamMember(_account, _isMember);
+    }
+
+    /// @notice An owner method for updating minAuthorBal.
+    /// @param  _amount new min balance of a poll author.
+    function updateMinAuthorBal(uint256 _amount) external onlyOwner() {
+        minAuthorBal = _amount;
+    }
+
+    /// @notice An owner method for updating maxPollsPerAuthor.
+    /// @param  _limit amount of active polls an author can have at any given time.
+    function updateMaxPollsPerAuthor(uint8 _limit) external onlyOwner() {
+        maxPollsPerAuthor = _limit;
+    }
+
+    // NOTE: governanceTokenAddr
+
+    /// @notice A method for updating team balance.
+    /// @param  _amount amount of BNB to add to teamBalance.
+    /// @dev    Only callable by governanceTokenAddr
+    function updateTeamBalance(uint256 _amount) external {
+        require(msg.sender == governanceTokenAddr, "Not Authorized");
+        teamBalance += _amount;
+    }
+
+    /// @notice A method for updating marketing balance.
+    /// @param  _amount amount of BNB to add to marketingBalanace.
+    /// @dev    Only callable by governanceTokenAddr
+    function updateMarketingBalance(uint256 _amount) external {
+        require(msg.sender == governanceTokenAddr, "Not Authorized");
+        marketingBalance += _amount;
+    }
+
+    // --------
+    // Internal
+    // --------
+
+    /// @notice Internal function for executing a poll.
+    /// @param _pollNum Unique poll number.
+    function _executeProposal(uint256 _pollNum) internal {
+
+        _updateEndTime(_pollNum);
```

Comment:
When the poll ends or is executed, the endTime becomes historical data where we can check to see  (instead of the proposed expiration date), when it actually ended or was executed